rules/common.smk

```diff
@@ -1,17 +1,20 @@
 import pandas as pd
+from peppy import SnakeProject
 from snakemake.utils import validate
 
-report: "../report/workflow.rst"
 
 ###### Config file and sample sheets #####
-configfile: "config.yaml"
+p = SnakeProject("prjcfg_native.yaml")
+configfile: p.snake_config
 validate(config, schema="../schemas/config.schema.yaml")
 
-samples = pd.read_table(config["samples"]).set_index("sample", drop=False)
+samples = p.sample_table
+
 validate(samples, schema="../schemas/samples.schema.yaml")
 
-units = pd.read_table(config["units"], dtype=str).set_index(["sample", "unit"], drop=False)
+units = p.subsample_table
```

Comment:
Really nice! What about always converting the subsample_table index into string inside peppy (in order to get rid of the line below)?

Reply:
Hi johannes, it looks like this has been done (the line is now gone).